### PR TITLE
fix: upgrade permission-controller to 8.0.0

### PR DESCRIPTION
**Pull request by frankvonhoven**
## **Description**

Upgrades the `permission-controller` from 7.1.0 --> 8.0.0

## **Related issues**

Fixes: [1695](https://github.com/MetaMask/mobile-planning/issues/1695)

## **Manual testing steps**

* Connect to app
    * Single account
    * Multiple accounts
* With multiple connected accounts
    * Disconnect Account
        * Next connected account should be selected
* Open a new tab
    * Connect to test app
    * https://metamask.github.io/test-dapp
        * Connect accounts
        * Disconnect account

### Connect Dapp Single Account

https://github.com/MetaMask/metamask-mobile/assets/141057783/9ca0afd9-5b61-4f2d-ba12-e7b1bd12e3b4

### Connect Dapp Multiple Accounts

https://github.com/MetaMask/metamask-mobile/assets/141057783/49b1d67c-6b2e-4203-b172-53420a8753d4

### TestDapp Connect Account

https://github.com/MetaMask/metamask-mobile/assets/141057783/fa896121-34b7-4c2f-8680-6e85640e60cf

### TestDapp Connect/Disconnnect/GET Account

https://github.com/MetaMask/metamask-mobile/assets/141057783/ded58259-4679-4790-85f7-0a34a3b51ca3

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -1028,7 +1028,7 @@ EXTERNAL SOURCES:
 
 SPEC CHECKSUMS:
   Base64: cecfb41a004124895a7bcee567a89bae5a89d49b
-  boost: 7dcd2de282d72e344012f7d6564d024930a6a440
+  boost: 57d2868c099736d80fcd648bf211b4431e51a558
```
Comment (legobeat) [inline ios/Podfile.lock:1031]: Why is this change happening here?

File: yarn.lock [hunk 2]
```diff
@@ -4294,6 +4279,21 @@
     immer "^9.0.6"
     nanoid "^3.1.31"
 
+"@metamask/permission-controller@^7.1.0":
```
Comment (legobeat) [inline yarn.lock:4282]: Prior to this PR, the `@metamask/snaps-utils` of `@metamask/accounts-controller` is using the same version of `#metamask/permission-controller` as the direct dependency of metamask-mobile.

In this PR, they are now different as `@metamask/accounts-controller` is kept at v10.

Is this fine?

Updating to `@metamask/accounts-controller` v13 would align it with the upgrade here.

File: ios/Podfile.lock (hunk 1 above)
Comment (frankvonhoven) [reply ios/Podfile.lock]: Accidental inclusion - fixed 👍

File: yarn.lock (hunk 2 above)
Comment (Cal-L) [reply yarn.lock:4282]: Based on what is used by `snaps-utils` - `SubjectType`, `SubjectPermissions`, and `PermissionConstraint`, should be the same between v7.1.0 and v8.0.0 of PermissionController, so we should be fine. @frankvonhoven Could you double check this as well?

Comment (frankvonhoven) [reply yarn.lock:4282]: Yep, looks like it's only using `permission-controller` for typings, we should be good

**Comment by legobeat** [comment]
LGTM!

(Needs a rebase on `main` to resolve the conflicting `yarn.lock`, though)

File: yarn.lock [hunk 3]
```diff
@@ -4241,6 +4241,7 @@
     "@metamask/safe-event-emitter" "^2.0.0"
     through2 "^2.0.3"
 
+<<<<<<< HEAD
```
Comment (legobeat) [inline yarn.lock:4244]: Merge conflict in lockfile. To resolve (assuming synced `main` branch): `git checkout main yarn.lock && rm -rf node_modules && yarn && yarn deduplicate

Comment (legobeat) [reply yarn.lock:4244]: NVM, somehow GH gave me a notification pointing to the old commit where it looked like this wasn't resolved already, which it is.

**Review by tommasini — APPROVED** [review]
LGTM!

**Pull request merged**